File: frontend/types/audit.ts
export interface AuditLog {
  id: string;

  merchant_id: string;

  recommendation_id: string;

  event_type: string;

  actor: string;

  details: string;

  created_at: string;
}

export interface AuditSummary {
  total: number;

  approved: number;

  rejected: number;

  executed: number;
}

export interface AuditDashboard {
  summary: AuditSummary;

  logs: AuditLog[];
}

File: backend/app/schemas/audit.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel
from pydantic import ConfigDict


class AuditLogResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: UUID

    merchant_id: UUID

    action: str

    reasoning: str

    status: str

    created_at: datetime


class AuditSummaryResponse(BaseModel):
    total: int

    approved: int

    rejected: int

    executed: int


class AuditDashboardResponse(BaseModel):
    summary: AuditSummaryResponse

    logs: list[AuditLogResponse]